File: app/src/main/java/com/example/moneymanager/MainActivity.java
package com.example.moneymanager;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.app.Activity;

public class MainActivity extends Activity {

    EditText editText;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d("LIFECYCLE", "MainActivity: onCreate");

        editText = findViewById(R.id.editText);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = editText.getText().toString();
                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                intent.putExtra("my_text", text);
                startActivity(intent);
            }
        });

        if (savedInstanceState != null) {
            String savedText = savedInstanceState.getString("saved_text");
            if (savedText != null) {
                editText.setText(savedText);
            }
        }
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("saved_text", editText.getText().toString());
        Log.d("LIFECYCLE", "MainActivity: сохранено");
    }

    @Override
    protected void onStart() { super.onStart(); Log.d("LIFECYCLE", "MainActivity: onStart"); }
    @Override
    protected void onResume() { super.onResume(); Log.d("LIFECYCLE", "MainActivity: onResume"); }
    @Override
    protected void onPause() { super.onPause(); Log.d("LIFECYCLE", "MainActivity: onPause"); }
    @Override
    protected void onStop() { super.onStop(); Log.d("LIFECYCLE", "MainActivity: onStop"); }
    @Override
    protected void onDestroy() { super.onDestroy(); Log.d("LIFECYCLE", "MainActivity: onDestroy"); }
}

File: app/src/main/java/com/example/moneymanager/MainActivity.kt
package com.example.moneymanager

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.moneymanager.ui.theme.MoneyManagerTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            MoneyManagerTheme {
                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
                    Greeting(
                        name = "User",
                        modifier = Modifier.padding(innerPadding)
                    )
                }
            }
        }
    }
}

@Composable
fun Greeting(name: String, modifier: Modifier = Modifier) {
    Text(
        text = "Привет $name! Я помогу с менеджментом твоего кошелька ;)",
        modifier = modifier
    )
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview() {
    MoneyManagerTheme {
        Greeting("Android")
    }
}